drivers/storage/azure/storage/azure_storage.go

```diff
@@ -0,0 +1,914 @@
+	if d.tag() != "" {
+		return d.tag() + azure.TagDelimiter + name
+	}
+	return name
+}
+
+// Retrieve config arguments
+func (d *driver) getSubscriptionID() string {
+	if result := os.Getenv("AZURE_SUBSCRIPTION_ID"); result != "" {
+		return result
+	}
+	return d.config.GetString(azure.ConfigAZURESubscriptionIDKey)
+}
+
+func (d *driver) getResourceGroup() string {
+	if result := os.Getenv("AZURE_RESOURCE_GROUP"); result != "" {
+		return result
+	}
+	return d.config.GetString(azure.ConfigAZUREResourceGroupKey)
+}
+
+func (d *driver) getTenantID() string {
+	if result := os.Getenv("AZURE_TENANT_ID"); result != "" {
+		return result
+	}
+	return d.config.GetString(azure.ConfigAZURETenantIDKey)
+}
+
+func (d *driver) getStorageAccount() string {
+	if result := os.Getenv("AZURE_STORAGE_ACCOUNT"); result != "" {
+		return result
+	}
+	return d.config.GetString(azure.ConfigAZUREStorageAccountKey)
+}
+
+func (d *driver) getStorageAccessKey() string {
+	if result := os.Getenv("AZURE_STORAGE_ACCESS_KEY"); result != "" {
+		return result
+	}
+	return d.config.GetString(azure.ConfigAZUREStorageAccessKeyKey)
+}
+
+func (d *driver) getContainer() string {
+	if result := os.Getenv("AZURE_CONTAINER"); result != "" {
+		return result
+	}
+	if result := d.config.GetString(azure.ConfigAZUREContainerKey); result != "" {
+		return result
+	}
+	return "vhds"
+}
+
+func (d *driver) getClientID() string {
+	if result := os.Getenv("AZURE_CLIENT_ID"); result != "" {
+		return result
+	}
+	return d.config.GetString(azure.ConfigAZUREClientIDKey)
+}
+
+func (d *driver) getClientSecret() string {
+	if result := os.Getenv("AZURE_CLIENT_SECRET"); result != "" {
+		return result
+	}
+	return d.config.GetString(azure.ConfigAZUREClientSecretKey)
+}
+
+func (d *driver) getCertPath() string {
+	if result := os.Getenv("AZURE_CERT_PATH"); result != "" {
+		return result
+	}
+	return d.config.GetString(azure.ConfigAZURECertPathKey)
+}
+
+func (d *driver) getUseHTTPS() bool {
+	if result := os.Getenv("AZURE_USE_HTTPS"); result != "" {
+		return result == "true"
+	}
+	return d.config.GetBool(azure.ConfigAZUREUseHTTPSKey)
+}
+
+func (d *driver) tag() string {
+	return d.config.GetString(azure.ConfigAZURETagKey)
+}
+
+// TODO rexrayTag
+/*func (d *driver) rexrayTag() string {
+  return d.config.GetString("azure.rexrayTag")
+}*/
+
+var errGetLocDevs = goof.New("error getting local devices from context")
+
+func (d *driver) toTypesVolume(
+	ctx types.Context,
+	blobs *[]blobStorage.Blob,
+	attachments types.VolumeAttachmentsTypes) ([]*types.Volume, error) {
+
+	/* TODO:
+	var (
+		ld *types.LocalDevices
+		ldOK bool
+	)
+
+	if attachments.Devices() {
+		// Get local devices map from context
+		if ld, ldOK = context.LocalDevices(ctx); !ldOK {
+			return nil, errGetLocDevs
+		}
+	}
+	*/
+
+	var volumesSD []*types.Volume
+	for _, blob := range *blobs {
+		volumeSD, error := d.toTypeVolume(ctx, &blob, attachments)
+		if error != nil {
+			ctx.WithError(error).Error("Failed to convert volume")
```

Comment:
Hi @Andrey-mp,

Do you intend to have this error logged but not returned? If so perhaps log it as a warning instead?